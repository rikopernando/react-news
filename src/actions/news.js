export const setNews = (news) => ({
  type: "SET_NEWS",
  news
})

export const startSetNews = () => {
  return (dispatch) => {
    return fetch('https://newsapi.org/v2/top-headlines?country=id&apiKey=faa93d5c650945a6a34a760fac8c7f90').then((response) => {
      return response.json()
    })
    .then(({articles}) => {
      dispatch(setNews(articles))
    })
  }
}
