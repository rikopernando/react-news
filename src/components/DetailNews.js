import React from 'react'
import { connect } from 'react-redux'
import moment from 'moment'

const DetailNews = ({ news }) => (
    <div className="content-container">
        <p className="list-news__author">
            { news.author ? news.author : 'Anonymous' } - { moment(news.publishedAt).format('D/M/YYYY') }
        </p>
        <h2>{news.title}</h2>
        <p style={{ wordSpacing: 5, letterSpacing: 1 }}>{news.content}</p>
        <a href={news.url} target="_blank">lihat asli</a>
    </div>
)

const mapStateToProps = (state,props) => ({
  news: state.news[props.match.params.id]
})

export default connect(mapStateToProps)(DetailNews)
