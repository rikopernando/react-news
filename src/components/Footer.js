import React from 'react'

const Footer = () => (
  <footer>
      <div className="content-container">
          <div className="footer-bar">
              <div>Kemang, Jakarta Selatan</div>
              <div>pernandoriko3.gmail.com</div>
          </div>
      </div>
  </footer>
)

export default Footer
