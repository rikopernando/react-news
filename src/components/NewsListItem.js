import React from 'react'
import moment from 'moment'
import { LazyLoadImage } from 'react-lazy-load-image-component'


export const NewsListItem = ({ urlToImage, title, description, author, publishedAt }) => (
  <div className="list-news">
    <LazyLoadImage
      className="list-news__image"
      src={ urlToImage ? urlToImage : './images/bg.jpg' }
      effect="blur"
    />
    <div>
      <h3 className="list-news__title">{ title }</h3>
      <p className="list-news__description">
        {
          (description.length > 125) ? description.substr(0,125) + '...' : description
        }
      </p>
      <p className="list-news__author">
        { author ? author : 'Anonymous' } - { moment(publishedAt).format('D/M/YYYY') }
      </p>
    </div>
  </div>
)

export default NewsListItem

