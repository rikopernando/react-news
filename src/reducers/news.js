const newsReducersDefaultState = []

export default (state = newsReducersDefaultState, actions) => {
  switch (actions.type) {
    case "SET_NEWS":
      return actions.news
    default:
      return state
  }
}
