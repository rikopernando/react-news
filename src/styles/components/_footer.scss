footer {
    padding: 20px 0;
    border-top: 3px solid $off-white;
    margin: 30px auto;
    max-width: 80rem;
}

footer .footer-bar {
    display: flex;
    justify-content: space-between;
    color: $grey;
}
