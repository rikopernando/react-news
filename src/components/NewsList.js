import React from 'react'
import { connect } from 'react-redux'
import NewsListItem from './NewsListItem'

export const NewsList = ({ news }) => (
  <div className="content-container">
    {
      news.map((news, key) => {
        return (
          <NewsListItem key={key} {...news} />
        )
      })
    }
  </div>
)

const mapStateToProps = (state) => ({
  news: state.news
})

export default connect(mapStateToProps)(NewsList)
