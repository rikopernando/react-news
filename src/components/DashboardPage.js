import React from 'react'
import NewsList from './NewsList'

const DashboardPage = () => (
  <div>
    <div className="content-container">
      <h2>
        Indonesia news 
      </h2> 
      <hr />
    </div>
    <NewsList />
  </div>
)

export default DashboardPage

