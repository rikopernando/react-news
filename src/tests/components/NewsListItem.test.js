import React from 'react'
import { shallow } from 'enzyme'
import { NewsListItem } from '../../components/NewsListItem'
import news from '../fixtures/news'

test('should rendered NewsListItem correctyly', () => {
  const wrapper = shallow(<NewsListItem {...news[1]} />)
  expect(wrapper).toMatchSnapshot()
})
