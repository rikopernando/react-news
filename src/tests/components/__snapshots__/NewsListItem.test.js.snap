// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered NewsListItem correctyly 1`] = `
<div
  className="list-news"
>
  <t
    afterLoad={[Function]}
    beforeLoad={[Function]}
    className="list-news__image"
    delayMethod="throttle"
    delayTime={300}
    effect="blur"
    placeholderSrc=""
    src="http://cdn2.tstatic.net/jateng/foto/bank/images/pengidap-hiv-aids-anak-ditolak-sekolah.jpg"
    threshold={100}
    visibleByDefault={false}
    wrapperClassName=""
  />
  <div>
    <h3
      className="list-news__title"
    >
      Ganjar Minta 14 Anak Pengidap HIV/Aids yang Ditolak Sekolah Temui Wali Kota Solo - Tribun Jateng
    </h3>
    <p
      className="list-news__description"
    >
      Kasus 14 anak pengidap HIV/AIDS (ADHA) yang ditolak sekolah di SDN Purwotomo Solo mendapat perhatian serius Gubernur Jawa Ten...
    </p>
    <p
      className="list-news__author"
    >
      faisal affan
       - 
      16/2/2019
    </p>
  </div>
</div>
`;
